From 452cbcd17549477e0fdeb7e90b645d09828e4354 Mon Sep 17 00:00:00 2001
From: Chinh Nguyen <chinh.nguyen.jg@renesas.com>
Date: Wed, 13 Mar 2019 19:10:45 +0700
Subject: [PATCH 096/175] iommu/ipmmu-vmsa: Invalidate TLB when IPMMU
 translation error occurred

Invalidate TLB when IPMMU translation error occurred
as described in hardware manual.

Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Signed-off-by: Chinh Nguyen <chinh.nguyen.jg@renesas.com>
---
 drivers/iommu/ipmmu-vmsa.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/iommu/ipmmu-vmsa.c b/drivers/iommu/ipmmu-vmsa.c
index d49afd7..3493308 100644
--- a/drivers/iommu/ipmmu-vmsa.c
+++ b/drivers/iommu/ipmmu-vmsa.c
@@ -555,6 +555,9 @@ static irqreturn_t ipmmu_domain_irq(struct ipmmu_vmsa_domain *domain)
 	if (!(status & (IMSTR_PF | IMSTR_TF)))
 		return IRQ_NONE;
 
+	/* Flush the TLB as required when IPMMU translation error occurred. */
+	ipmmu_tlb_invalidate(domain);
+
 	/*
 	 * Try to handle page faults and translation faults.
 	 *
-- 
2.7.4

